core/sql/optimizer/NormRelExpr.cpp

```diff
@@ -3062,6 +3071,35 @@ GroupByAgg* Join::pullUpGroupByTransformation(NormWA& normWARef)
     if (CmpCommon::getDefault(SUBQUERY_UNNESTING) == DF_DEBUG)
       *CmpCommon::diags() << DgSqlCode(2997)   
 	<< DgString1("Subquery was not unnested. Reason: Left child does not have a unique constraint"); 
+
+    // Things to consider (referring to the picture above): If the all of the 
+    // following are true:
+    //   * {pred2} has only equals/VEG predicates of the form X.col = Y.col
+    //   * {aggr} does not have any outer references
+    //   * {pred3} does not have any outer references
```

Comment:
Lack of outer references in {aggr} and {pred3} may indicate that this transformation may not apply when we more than one 1 level of correlated subqueries. This set of conditions clearly explains when this transformation can be applied. I hope we can add the change needed for this soon. The description here also indicates that NULL preserving semantics is not affected by this new transformation. So we could use this change for OR subqueries where the Join could be a Left Join.